modules/budget/main.tf

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  project_name  = length(var.projects) == 0 ? "All Projects" : var.projects[0]
+  display_name  = var.display_name == null ? "Budget For ${local.project_name}" : var.display_name
+  pubsub_topics = var.alert_pubsub_topic == null ? [] : [var.alert_pubsub_topic]
+  projects = length(var.projects) == 0 ? null : [
+    for id in var.projects :
+    "projects/${id}"
+  ]
+  services = var.services == null ? null : [
+    for id in var.services :
+    "services/${id}"
+  ]
+}
+
+resource "google_billing_budget" "budget" {
+  provider = google-beta
+  count    = var.create_budget ? 1 : 0
+
+  billing_account = var.billing_account
+  display_name    = local.display_name
+
+  budget_filter {
+    projects               = local.projects
```

Comment:
A bit strange, the provider docs say this only support a 0 or 1 projects array currently 
https://www.terraform.io/docs/providers/google/r/billing_budget.html#projects

However, passing in more than one works just fine. The test I created does just that. When setting more than one it also shows up in the billing budget UI on the console just fine too. So seems like the provider docs are just a bit out of date.